test/adapter/json/collection_test.rb

```diff
@@ -85,6 +85,20 @@ def test_include_option
 
             assert_equal(expected, actual)
           end
+
+          def test_fields_with_no_associations_include_option
+            serializer = ArraySerializer.new([@first_post, @second_post])
+            adapter = ActiveModel::Serializer::Adapter::Json.new(serializer)
+            actual = adapter.serializable_hash(fields: [:id])
```

Comment:
@bf4 Well currently the Json adapter *does* support a lesser version of the `fields` option (namely it allows to specify the attributes but just for the primary data). Support is broken though.